Feature/fill panel cote

kev43 opened the pull request:
Affichage des infos des joueurs sur le coté. Toute la partie style est faite, il faut juste interconnecter avec le back.

kev43 committed:
Ajout fonction rempissage avec infos des joueurs sur les cotés.
Il faut lier ces infos avec les données du Back

kev43 committed:
Gestion de l'afficha des Panel sur les cotés améliorée

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -79,14 +87,16 @@ public QuoridorGUI(String title, int taille) {
         garageMurGauche.setPreferredSize(new Dimension(tailleLargeurGarageMur , tailleIHMHauteur));
         garageMurGauche.setBounds(0, 0, tailleLargeurGarageMur, tailleIHMHauteur);
-        garageMurGauche.setBackground(Color.GREEN);
+        //garageMurGauche.setBackground(Color.GREEN);
```
Enleve les commentaires inutiles

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -79,14 +87,16 @@ public QuoridorGUI(String title, int taille) {
         garageMurDroit.setPreferredSize(new Dimension(tailleLargeurGarageMur , tailleIHMHauteur));
         garageMurDroit.setBounds(taillePlateauQuoridor+tailleLargeurGarageMur, 0, tailleLargeurGarageMur, tailleIHMHauteur);
-        garageMurDroit.setBackground(Color.BLUE);
+        //garageMurDroit.setBackground(Color.BLUE);
```
commentaire a enlever

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -153,9 +163,12 @@ public QuoridorGUI(String title, int taille) {
             }
         }
 
+        //plateauQuoridor.setBorder(BorderFactory.createLineBorder(Color.BLACK ,2));
```
commentaire a enlever

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -224,11 +237,12 @@ public void mouseReleased(MouseEvent e) {
                     parent.add(pion);
                     pion.setVisible(true);
-                    update();
+                    //update();
```
commentaire a enlever

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -224,11 +237,12 @@ public void mouseReleased(MouseEvent e) {
                 }
                 else {
-                    update();
+                    //update();
```
commentaire a enlever

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -440,4 +454,74 @@ private void affichePion(Coordonnees coord, Couleur c) {
     }
+
+
+    private void remplissagePanelCote (/*TODO passer la liste des joueurs IHM*/) {
```
Je ne peux pas accepter ta PR avec des TODO, il faudra que tu y fasse les modifications pour enlever les TODO

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -440,4 +454,74 @@ private void affichePion(Coordonnees coord, Couleur c) {
+        int nbRowInColumn = 0;
+
+
+        garageMurDroit.setBackground(Color.PINK);
```
Pourquoi changer la couleur du background ligne 98 pour les re changer maintenant? Un des 2 est de trop

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -440,4 +454,74 @@ private void affichePion(Coordonnees coord, Couleur c) {
+    private void remplissagePanelCote (/*TODO passer la liste des joueurs IHM*/) {
+        /*TODO Count nombre de joueurs pour les afficher sur les cotés*/
+        int nbPlayer = 4 ; //Utiliser la liste des joueurs au lieu de la valmeur en dur
+        int nbRowInColumn = 0;
```
Que représente exactement cette int? N'hesite pas a le mettre en commentaire

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -440,4 +454,74 @@ private void affichePion(Coordonnees coord, Couleur c) {
+        tabPanelGauche = new JPanel[nbRowInColumn][1];
+        for (int i =0 ; i<nbRowInColumn ; i++) {
+            tabPanelGauche[i][0] = new JPanel();
+            tabPanelGauche[i][0].setBackground(new Color(404040));
```
Pourquoi utiliser une couleur en hexa maintenant? Au minimum précise la couleur en commentaire
Ici pour un gris, je ne pense pas que ce soit nécessaire de passer par l'hexa

kev43 replied:
Utilisation de code hexa, car plus nuances non disponnibles avec des macros de type Color.RED.
J'ajoute un comentaire avec la couleur

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -440,4 +454,74 @@ private void affichePion(Coordonnees coord, Couleur c) {
+    }
+
+    private void createLabelCote (JPanel jp, int numJoueur) {
+        /*TODO GetCurrentPlayer*/
```
IDEM, TODO a faire avant merge

kev43 committed:
Modifs pour PR + ajustement de la position de la fenetre

kev43 commented:
J'ai fait les modifs.
ça doit être bon

kev43 commented:
J'ai fait les modifs demandées.

Le 22 décembre 2017 à 11:14, DuflotFlorian <notifications@github.com> a
écrit :

> *@DuflotFlorian* requested changes on this pull request.
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158461944>
> :
>
> > @@ -79,14 +87,16 @@ public QuoridorGUI(String title, int taille) {
>          layeredPane.add(garageMurGauche, JLayeredPane.DEFAULT_LAYER);
>          garageMurGauche.setPreferredSize(new Dimension(tailleLargeurGarageMur , tailleIHMHauteur));
>          garageMurGauche.setBounds(0, 0, tailleLargeurGarageMur, tailleIHMHauteur);
> -        garageMurGauche.setBackground(Color.GREEN);
> +        //garageMurGauche.setBackground(Color.GREEN);
>
> Enleve les commentaires inutiles
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158461971>
> :
>
> >
>          //Creation du stockage à murs gauche
>          garageMurDroit = new JPanel();
>          layeredPane.add(garageMurDroit, JLayeredPane.DEFAULT_LAYER);
>          garageMurDroit.setPreferredSize(new Dimension(tailleLargeurGarageMur , tailleIHMHauteur));
>          garageMurDroit.setBounds(taillePlateauQuoridor+tailleLargeurGarageMur, 0, tailleLargeurGarageMur, tailleIHMHauteur);
> -        garageMurDroit.setBackground(Color.BLUE);
> +        //garageMurDroit.setBackground(Color.BLUE);
>
> commentaire a enlever
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158462008>
> :
>
> > @@ -153,9 +163,12 @@ public QuoridorGUI(String title, int taille) {
>              }
>          }
>
> +        //plateauQuoridor.setBorder(BorderFactory.createLineBorder(Color.BLACK ,2));
>
> commentaire a enlever
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158462037>
> :
>
> > @@ -224,11 +237,12 @@ public void mouseReleased(MouseEvent e) {
>                      Container parent = (Container) cmp;
>                      parent.add(pion);
>                      pion.setVisible(true);
> -                    update();
> +                    //update();
>
> commentaire a enlever
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158462053>
> :
>
> >                  }
>                  else {
> -                    update();
> +                    //update();
>
> commentaire a enlever
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158462316>
> :
>
> > @@ -440,4 +454,74 @@ private void affichePion(Coordonnees coord, Couleur c) {
>          JLabel piece = new JLabel(new ImageIcon(imageURL));
>          j.add(piece);
>      }
> +
> +
> +    private void remplissagePanelCote (/*TODO passer la liste des joueurs IHM*/) {
>
> Je ne peux pas accepter ta PR avec des TODO, il faudra que tu y fasse les
> modifications pour enlever les TODO
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158462474>
> :
>
> > @@ -440,4 +454,74 @@ private void affichePion(Coordonnees coord, Couleur c) {
>          JLabel piece = new JLabel(new ImageIcon(imageURL));
>          j.add(piece);
>      }
> +
> +
> +    private void remplissagePanelCote (/*TODO passer la liste des joueurs IHM*/) {
> +        /*TODO Count nombre de joueurs pour les afficher sur les cotés*/
> +        int nbPlayer = 4 ; //Utiliser la liste des joueurs au lieu de la valmeur en dur
> +        int nbRowInColumn = 0;
> +
> +
> +        garageMurDroit.setBackground(Color.PINK);
>
> Pourquoi changer la couleur du background ligne 98 pour les re changer
> maintenant? Un des 2 est de trop
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158462581>
> :
>
> > @@ -440,4 +454,74 @@ private void affichePion(Coordonnees coord, Couleur c) {
>          JLabel piece = new JLabel(new ImageIcon(imageURL));
>          j.add(piece);
>      }
> +
> +
> +    private void remplissagePanelCote (/*TODO passer la liste des joueurs IHM*/) {
> +        /*TODO Count nombre de joueurs pour les afficher sur les cotés*/
> +        int nbPlayer = 4 ; //Utiliser la liste des joueurs au lieu de la valmeur en dur
> +        int nbRowInColumn = 0;
>
> Que représente exactement cette int? N'hesite pas a le mettre en
> commentaire
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158462790>
> :
>
> > +        JPanel [][] tabPanelDroit = null;
> +
> +        if (nbPlayer == 2) {
> +            nbRowInColumn = 3;
> +
> +        } else if (nbPlayer == 4) {
> +            nbRowInColumn = 5;
> +        }
> +
> +        garageMurGauche.setLayout(new GridLayout(nbRowInColumn,1));
> +        garageMurDroit.setLayout(new GridLayout(nbRowInColumn,1));
> +
> +        tabPanelGauche = new JPanel[nbRowInColumn][1];
> +        for (int i =0 ; i<nbRowInColumn ; i++) {
> +            tabPanelGauche[i][0] = new JPanel();
> +            tabPanelGauche[i][0].setBackground(new Color(404040));
>
> Pourquoi utiliser une couleur en hexa maintenant? Au minimum précise la
> couleur en commentaire
> Ici pour un gris, je ne pense pas que ce soit nécessaire de passer par
> l'hexa
> ------------------------------
>
> In src/vue/QuoridorGUI.java
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#discussion_r158462867>
> :
>
> > +        }
> +
> +        if(nbPlayer==2) {
> +            createLabelCote(tabPanelGauche[1][0],1);
> +            createLabelCote(tabPanelDroit[1][0],2);
> +
> +        } else if (nbPlayer == 4){
> +            createLabelCote(tabPanelGauche[1][0],1);
> +            createLabelCote(tabPanelDroit[1][0],2);
> +            createLabelCote(tabPanelGauche[3][0],3);
> +            createLabelCote(tabPanelDroit[3][0],4);
> +        }
> +    }
> +
> +    private void createLabelCote (JPanel jp, int numJoueur) {
> +        /*TODO GetCurrentPlayer*/
>
> IDEM, TODO a faire avant merge
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DuflotFlorian/Quoridor_4IRC/pull/27#pullrequestreview-85277847>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFepaR4MkwhMuZFijDLxybmC_GOhFexAks5tC4EXgaJpZM4RKXV6>
> .
>

DuflotFlorian commented on src/vue/QuoridorGUI.java:
```diff
@@ -237,10 +234,12 @@ public void mouseReleased(MouseEvent e) {
                     pion.setVisible(true);
-                    //update();
+
+                    update(); //Ligne a commenter pour debug, permet au pion de ne pas toujours revenir à sa position intiale
```
Cela n'a aucun sens. Ca veut dire que sur la version fonctionnelle, le pion reviens toujours à sa position initiale?

kev43 replied:
Done. Ce n'était plus utile

DuflotFlorian commented:
Quand je dit d'enlever un TODO ca ne veut pas dire de le supprimer, mais de le faire...

kev43 commented:
Les TODO sont faits.
Pour faire cela j'ai du merge la branche update2.0 dans ma branche car j'avais besoin des fonctions du back pour faire quelque chose de fonctionnel.

La gestion du nombre de murs affiché sera mis à jour avec la gestion des murs coté back.

DuflotFlorian commented:
Ba non Kevin, il ne faut pas faire ca.
Regarde ce site https://danielkummer.github.io/git-flow-cheatsheet/
Il t'expliquera comment utiliser de la bonne manière les branches Git

kev43 committed:
Reverse des mauvais commits.
Rebase sur la branch develop actuelle pour faciliter le merge.
Ajout des derniers ajustements.
Normalement Prêt à être merge sur develop

kev43 committed:
Suppression ligne commentée inutile

kev43 commented:
J'ai revert mon merge.
J'ai rebase mes commits à partir du devellop actuel pour obtenir quelque chose de testable.

Normalement ça peut se merge dans develop.
Ce coup-ci je pense avoir tout vérifié et ne pas avoir fait d'erreur à cause de mon niveau moisi en Git.

DuflotFlorian committed:
Merge branch 'develop' into Feature/FillPanelCote
